polar-core/src/vm.rs

```diff
@@ -1889,25 +1969,185 @@ impl PolarVirtualMachine {
                 ))
             }
         }
-        Ok(())
+        Ok(QueryEvent::None)
     }
 
+    /// Handle variables & constraints as arguments to various operations.
+    /// Calls the `eval` method to handle ground terms.
+    ///
+    /// Arguments:
+    ///
+    /// - handle_unbound_left_var: If set to `false`, allow `eval` to handle
+    ///   operations with an unbound left variable, instead of adding a constraint.
+    ///   Some operations, like `In`, emit new goals or choice points when the left
+    ///   operand is a variable.
+    /// - handle_unbound_right_var: Same as above but for the RHS. `Dot` uses this.
     #[allow(clippy::many_single_char_names)]
-    fn in_op_helper(
+    fn query_op_helper<F>(
```

Comment:
Should be above `dot_op_helper()`

Reply:
Agree, didn't move it just to make the diff easier to read. Will move.